template.yaml

```diff
@@ -726,3 +780,38 @@ Outputs:
     Value: !GetAtt LambdaCodeUpdateHandler.Arn
     Export:
       Name: !Sub "${AWS::StackName}-CodeUpdateHandler"
+  UserPoolClientUpdateHandler:
+    Description: The User Pool clinet update handler's function ARN of the custom resource that adds updates the User Pool Configuration
+    Value: !GetAtt UserPoolClientUpdateHandler.Arn
+    Export:
+      Name: !Sub "${AWS::StackName}-UserPoolClientUpdateHandler"  
+  LambdaEdgeExecutionRole:
+    Description: The ARN of the lambda execution role associated with the edge functions. 
+    Value: !GetAtt LambdaEdgeExecutionRole.Arn
+    Export:
+      Name: !Sub "${AWS::StackName}-LambdaEdgeExecutionRole"          
+  LambdaCodeUpdateHandlerRole:
+    Description: The ARN of the lambda execution role associated with the edge functions. 
+    Value: !Ref LambdaCodeUpdateHandlerRole
+    Export:
+      Name: !Sub "${AWS::StackName}-LambdaCodeUpdateHandlerRole"          
+  LambdaCodeUpdateHandlerFunction:
+    Description: The ARN of the lambda execution role associated with the edge functions. 
+    Value: !GetAtt LambdaCodeUpdateHandler.Arn
+    Export:
+      Name: !Sub "${AWS::StackName}-LambdaCodeUpdateHandlerFunction"          
+  UserPoolOAuthScopes:
+    Description: The OAuthScopes used in the initial creation of the user pool. May need to be referenced again. 
+    Value: !Ref OAuthScopes
+    Export:
+      Name: !Sub "${AWS::StackName}-UserPoolOAuthScopes"
+  WebsiteUrls:
+    Description: The ${WebsiteUrl} combined with ${AlternateDomainNames}. Can be used for CORS policies
+    Condition: CreateCloudFrontDistribution
+    Value: 
+      Fn::Join: 
+        - ","
+        - - !Sub "https://${CloudFrontDistribution.DomainName}"
+          - !Join [',', !Ref AlternateDomainNames]
+    Export:
+      Name: !Sub "${AWS::StackName}-WebsiteUrls"
```

Comment:
These are from `authorization-lambda-at-edge`?

Can you explain this part a bit more?

Reply:
This is `authorization-lambda-at-edge`, so all of them are. I'm exporting all these values to be able to reuse them from the CF templates using this one (e.g. knowledge-vault, ASEC). Let's hear each other a bit or please try to rephrase the question so i know exactly what to focus on with the explanation.

Reply:
This is `cloudfront-authorization-at-edge` &ndash; I was wondering if you took these from the other AWS example ([`authorization-lambda-at-edge`](https://github.com/aws-samples/authorization-lambda-at-edge)).

But it's clear now; I see the references in https://github.com/studyportals/Aperture-Science-Enrichment-Center/pull/19/files#diff-642cdd44a2557892e63952d1e29fdff9.